src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -1492,7 +1494,7 @@ struct WindowFunctionLagLeadInFrame final : public WindowFunction
     }
 
     DataTypePtr getReturnType() const override
-    { return argument_types[0]; }
+    { return std::make_shared<DataTypeNullable>(argument_types[0]); }
```

Comment:
Are you also fixing the lagInFrame nullability problems in this PR? Better make two separate ones.

This change looks incorrect, because the return type of lagInFrame doesn't have to be nullable, e.g. when a not-null default value is specified. The problems in  https://github.com/ClickHouse/ClickHouse/issues/26115 are partly caused by that we apply nullability handling for aggregate functions to window functions as well, when we shouldn't. The fix needs some refactoring that would introduce deeper separation between aggregate and window functions. I started working on it but was distracted by other tasks. Probably will continue next week.

So in the meantime, could you please remove the fix for lagInFrame from this PR, and keep only the new function nth_value? That function looks good.

Reply:
Thanks for your comments, you are right, the force of nullability here is not appropriate and incomplete I accidently pushed it, I will clean it up, and I'm still looking into this problem. Speaking of the non-null default value, I found the result is not correct if the given default value is an arithmetic expression with type cast required, e.g. if the default value for the 3rd arg is like value*1.5, it looks like over flows. I tracked all the way up to interpreter, and also with null value, my guess there might be something in parser when it deduce the type of the 3rd arg? But I haven't looked into that part yet. I'm new in CH, still trying to understand lots of things.